src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -978,6 +980,90 @@ void goto_instrument_parse_optionst::instrument_goto_program()
   if(cmdline.isset("remove-function-pointers"))
     do_function_pointer_removal();
 
+  if(cmdline.isset("function-inline"))
+  {
+    std::string function=cmdline.get_value("function-inline");
+    assert(!function.empty());
+
+    do_function_pointer_removal();
+
+    status() << "Inlining calls of function `" << function << "'" << eom;
+
+    if(!cmdline.isset("log"))
+    {
+      goto_function_inline(
+        goto_functions,
+        function,
+        ns,
+        ui_message_handler,
+        true);
+    }
+    else
+    {
+      std::string filename=cmdline.get_value("log");
+      bool have_file=!filename.empty() && filename!="-";
+
+      jsont result=
+        goto_function_inline_and_log(
+          goto_functions,
+          function,
+          ns,
+          ui_message_handler,
+          true);
+
+      if(have_file)
+      {
+#ifdef _MSC_VER
+        std::ofstream of(widen(filename));
+#else
+        std::ofstream of(filename);
+#endif
+        if(!of)
+          throw "failed to open file "+filename;
+
+        of << result;
+        of.close();
+      }
+      else
+      {
+        std::cout << result << std::endl;
+      }
+    }
+
+    goto_functions.update();
+    goto_functions.compute_loop_numbers();
+  }
+
+  if(cmdline.isset("partial-inline"))
+  {
+    do_function_pointer_removal();
+
+    status() << "Partial inlining" << eom;
+    goto_partial_inline(goto_functions, ns, ui_message_handler, true);
+
+    goto_functions.update();
+    goto_functions.compute_loop_numbers();
+  }
```

Comment:
The inliner has an option ``adjust_function`` which indicates whether the ``function`` field of the inlined instructions should be updated to the value of the function they are being inlined into. The behavior of the previous inliner was to not change the function field. ``do_partial_inlining()`` still follows this old behavior, whereas here we do adjust the ``function`` field.

The reason why the ``function`` field has previously not been updated seems to have to do with how loop ids are formed. I would vote to always update the ``function`` field and make loop ids independent of that (as suggested in #295). What's your opinion on this?